### DISCUSSION TITLE
PatchInstall Parameter

### DISCUSSION THREAD

**Discussion by mikiekelly** [Esri/arcgis-powershell-dsc#539] [Ideas]
Sometimes as you are iterating through versions, and constantly reinstalling components, it is nice to keep code in your config file but turn on/off parts that streamline the operation. For this reason, I think it would be valuable to have a boolean PatchInstall parameter that dictates whether or not patches are installed. As things stand you either need to remove json from the config file, or delete/move patches downloaded from prior installs. This would allow you keep everything in place, with a minor edit to turn on/off the feature.

I'd also be satisfied if an empty PatchInstallOrder parameter ensured no patches were installed. But this doesn't appear to be the case.

**Comment by cameronkroeker** [comment]
@mikiekelly This is an excellent suggestion! We will look to get this added in an upcoming release of the module.

**ConfigData.SkipPatchInstalls** - boolean true|false (default). When set to false, and PatchesDir is defined the module will install patches from the PatchesDir directory, if set to true it will skip both download and installing of patches.

**Reply by mikiekelly** [reply]
Nice one sounds good to me. Thanks!

**Comment by cameronkroeker** [comment]
This has been implemented in the latest release, v4.3.0!

https://github.com/Esri/arcgis-powershell-dsc/releases/tag/v4.3.0

https://github.com/Esri/arcgis-powershell-dsc/blob/4e1ea910cf3c237c378dfcfc1130a296de3c4bcb/SampleConfigs/v4/v4.3.0/Base%20Deployment/BaseDeployment-SingleMachine.json#L32

Thanks,
Cameron K.

**Reply by mikiekelly** [reply]
Quite the turnaround! Thanks Cameron - that's fantastic. Appreciate your support.
